src/os_linux.cpp

```diff
@@ -388,4 +390,42 @@ int OS::mprotect(void* addr, size_t size, int prot) {
     return ::mprotect(addr, size, prot);
 }
 
+static int checkPreloadedCallback(dl_phdr_info* info, size_t size, void* data) {
+    Dl_info* dl_info = (Dl_info*)data;
+
+    Dl_info current_info = dl_info[0];
+    Dl_info sigaction_info = dl_info[1];
+
+    if (info->dlpi_name == NULL) {
+        return 0;
+    } else if (strcmp(info->dlpi_name, current_info.dli_fname) == 0) {
+        return 1;
+    } else if (strcmp(info->dlpi_name, sigaction_info.dli_fname) == 0) {
+        return -1;
+    }
+
+    return 0;
+}
+
+bool OS::checkPreloaded() {
+    if (getenv("LD_PRELOAD") == NULL) {
+        return false;
+    }
+
+    Dl_info current_info;
+    if (dladdr((const void*)OS::checkPreloaded, &current_info) == 0 || current_info.dli_fname == NULL) {
+        return false;
+    }
+
+    Dl_info sigaction_info;
+    if (dladdr((const void*)sigaction, &sigaction_info) == 0 || sigaction_info.dli_fname == NULL) {
```

Comment:
Just wondering why `sigaction`? Not the most obvious function in libc - may confuse someone who sees this for the first time. A short comment describing the idea of `checkPreloaded` would be useful anyway.